CodeEdit/Features/NavigatorSidebar/NavigatorSidebarToolbarBottom.swift

```diff
@@ -33,7 +33,17 @@ struct NavigatorSidebarToolbarBottom: View {
             Button("Add File") {
                 guard let folderURL = workspace.workspaceClient?.folderURL() else { return }
                 guard let root = try? workspace.workspaceClient?.getFileItem(folderURL.path) else { return }
-                root.addFile(fileName: "untitled") // TODO: use currently selected file instead of root
+                let newFile = root.addFile(fileName: "untitled") // TODO: use currently selected file instead of root
+
+                DispatchQueue.global(qos: .background).async {
```

Comment:
This nested dispatch should be changed to a single `DispatchQueue.main.async`. `DispatchQueue.main.async` will queue the completion for the next render loop, so there's no need to do two context switches.

Reply:
Good catch, thanks! I sent a new commit.